pkg/agent/agent.go

```diff
@@ -87,6 +92,23 @@ func (a *Agent) readPump() {
 		return a.conn.SetReadDeadline(time.Now().Add(pongWait))
 	})
 
+	a.conn.SetCloseHandler(func(err int, msg string) error {
+		a.logger.Debug("peer disconnected", logging.Fields{"room": a.room, "peer": a.peer.UID, "error_code": err, "error_msg": msg})
+		
+		jsonMessage, jsonError := json.Marshal(DisconnectPayload{"peer_disconnected", a.peer.UID})
+		if jsonError != nil {
+			a.logger.Error("can't marshal object to json")
+			return nil
+		}			
+
+		a.broker.Send(a.room, messaging.Message{
+			From:    "tarpon",
```

Comment:
I think we should use peer ID as from, not tarpon. Technically, there's no protection against some peer having ID set to "tarpon". What do you think? If we want to use something special, we should add a protection that peers can't be registered with "tarpon" ID. But maybe we can just use peer ID in the From field? Then we should probably remove peer id from the payload to avoid confusion on the client about which ID to use. "From" can't be spoofed by the clients, but the one in the payload can be, so non well behaving clients could send messages that some other client disconnected.